src/egraph/rebuild.rs

```diff
@@ -3,23 +3,24 @@ use crate::*;
 impl<L: Language, N: Analysis<L>> EGraph<L, N> {
     // proof.l should be i.
     // proof.r should be missing a few slots.
-    fn record_redundancy_witness(&mut self, i: Id, cap: &HashSet<Slot>, proof: ProvenEq) {
+    fn record_redundancy_witness(&mut self, i: Id, cap: &HashSet<Slot>, _proof: ProvenEq) {
```

Comment:
again, this reads a bit confusing as it seems to mean that the proof is not used in this function, which is not correct.
Maybe #[allow(unused)] can help, but not super important.